lib/publiccloud/gcp_client.pm

```diff
@@ -0,0 +1,162 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for google connection and authentication
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package publiccloud::gcp_client;
+use Mojo::Base -base;
+use testapi;
+use utils;
+use publiccloud::vault;
+use Mojo::Util qw(b64_decode);
+use Mojo::JSON 'decode_json';
+
+use constant CREDENTIALS_FILE => '/root/google_credentials.json';
+
+has account => undef;
+has project_id => undef;
+has private_key_id => undef;
+has private_key => undef;
+has service_acount_name => undef;
+has client_id => undef;
+has vault_gcp_role_index => undef;
+has gcr_zone => undef;
+has vault => undef;
+
+sub vault_gcp_roles {
+    return split(/\s*,\s*/, get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
+}
+
+sub init {
+    my ($self, %params) = @_;
+
+    $self->gcr_zone(get_var('PUBLIC_CLOUD_GCR_ZONE', 'eu.gcr.io'));
+
+    $self->vault(publiccloud::vault->new());
+
+    $self->create_credentials_file();
+    assert_script_run('source ~/.bashrc');
+    assert_script_run('ntpdate -s time.google.com');
+    assert_script_run('gcloud config set account ' . $self->account);
+    assert_script_run(
+        'gcloud auth activate-service-account --key-file=' . CREDENTIALS_FILE . ' --project=' . $self->project_id);
+}
+
+
+=head2
+A service account in GCP can only have up to 10 keys assigned. With this we 
+reach our paralel openqa jobs quite fast.
+To have more keys available, we create 4 service accounts and select randomly
+one. If this fails, the next call of C<get_next_vault_role()> will retrieve
+the next.
+
+=cut
+sub get_next_vault_role {
+    my ($self) = shift;
+    my @known_roles = $self->vault_gcp_roles();
+    if (defined($self->vault_gcp_role_index)) {
+        $self->vault_gcp_role_index(($self->vault_gcp_role_index + 1) % scalar(@known_roles));
+    }
+    else {
+        $self->vault_gcp_role_index(int(rand(scalar(@known_roles))));
+    }
+    return $known_roles[$self->vault_gcp_role_index];
+}
+
+sub create_credentials_file {
+    my ($self) = @_;
+    my $credentials_file;
+
+    if ($self->private_key_id()) {
+        $credentials_file
+          = "{"
+          . $/
+          . '"type": "service_account", '
+          . $/
+          . '"project_id": "'
+          . $self->project_id . '", '
+          . $/
+          . '"private_key_id": "'
+          . $self->private_key_id . '", '
+          . $/
+          . '"private_key": "'
+          . $self->private_key . '", '
+          . $/
+          . '"client_email": "'
+          . $self->service_acount_name . '@'
+          . $self->project_id
+          . '.iam.gserviceaccount.com", '
```

Comment:
```suggestion
          . '"client_email": "' . $self->service_acount_name . '@' . $self->project_id . '.iam.gserviceaccount.com", '
```
Please make the `$credentials_file` more compact.

Reply:
Perltidy is doing this. :(

Reply:
Ok, tidy was saying that the work was not done. I created a new commit use the gcp_client logic directly on the gce library